### DISCUSSION TITLE
AnsiTerminal feature request

### DISCUSSION THREAD

**Discussion by johngalt01** [fdivitto/FabGL#298] [Ideas]
Hi,

would if be possible to add a function to the Ansi Terminal such that you can send it a request for what color is located at a specific X,Y pixel position and it would return an integer value as a keypress from 0 to 255.

so that if i set a pixel at 100,100 color 16 for example then i can request the terminal return the value located at 100,100 later.
this would allow dumping the contents of the terminal screen back to a file so that it could be loaded later and restore what the contents of the screen are. you could use the terminal as photoshop this way or you can dump a section of the screen for recall later.

this is not in the scope of a normal terminal but would increase the functionality.

i suppose it would be something like10 PRINT CHR$(27)+"_RETURNPIXEL"+STR$(X%)+";"+STR$(Y%)+"$" : returnpixel$=Inkey$
 
it is a way to have DMA graphics in a way bidirectional communication.

**Comment by mobluse** [comment]
This could be a security hole. E.g. if you are logged in to some telnet site they could read pictures from your screen that you had before you logged in. I don't know if pictures and text are in the same layer, but if they are the site could also read text. Also, this functionality is unnecessary since if your program have written or drawn something on the users screen, you know what you have sent and it could be in the memory.

**Reply by johngalt01** [reply]
the issue is when you have 64KB of memory total making it impossible to keep the high resolution in memory, but you can dump the screen memory into a file and recall it later. maybe make something optional that a person can enable and take a risk on them.
